runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/io/DataflowDatastoreIOTest.java

```diff
@@ -1,66 +0,0 @@
-/*
- * Licensed to the Apache Software Foundation (ASF) under one
- * or more contributor license agreements.  See the NOTICE file
- * distributed with this work for additional information
- * regarding copyright ownership.  The ASF licenses this file
- * to you under the Apache License, Version 2.0 (the
- * "License"); you may not use this file except in compliance
- * with the License.  You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package org.apache.beam.runners.dataflow.io;
-
-import static org.apache.beam.sdk.transforms.display.DisplayDataMatchers.hasDisplayItem;
-
-import static org.hamcrest.Matchers.hasItem;
-import static org.junit.Assert.assertThat;
-
-import org.apache.beam.runners.dataflow.transforms.DataflowDisplayDataEvaluator;
-import org.apache.beam.sdk.io.datastore.DatastoreIO;
-import org.apache.beam.sdk.transforms.PTransform;
-import org.apache.beam.sdk.transforms.display.DisplayData;
-import org.apache.beam.sdk.transforms.display.DisplayDataEvaluator;
-import org.apache.beam.sdk.values.PBegin;
-import org.apache.beam.sdk.values.PCollection;
-import org.apache.beam.sdk.values.POutput;
-
-import com.google.datastore.v1beta3.Entity;
-import com.google.datastore.v1beta3.Query;
-
-import org.junit.Test;
-
-import java.util.Set;
-
-/**
- * Unit tests for Dataflow usage of {@link DatastoreIO} transforms.
- */
-public class DataflowDatastoreIOTest {
-  @Test
-  public void testSourcePrimitiveDisplayData() {
```

Comment:
Why is why I believe we should just leave the test as is and migrate it
when it can be versus deleting and forgetting about it.

On Fri, Jul 8, 2016 at 11:15 AM, Daniel Halperin <notifications@github.com>
wrote:

> In
> runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/io/DataflowDatastoreIOTest.java
> <https://github.com/apache/incubator-beam/pull/584#discussion_r70090410>:
>
> > -import org.apache.beam.sdk.values.PCollection;
> > -import org.apache.beam.sdk.values.POutput;
> > -
> > -import com.google.datastore.v1beta3.Entity;
> > -import com.google.datastore.v1beta3.Query;
> > -
> > -import org.junit.Test;
> > -
> > -import java.util.Set;
> > -
> > -/**
> > - * Unit tests for Dataflow usage of {@link DatastoreIO} transforms.
> > - */
> > -public class DataflowDatastoreIOTest {
> > -  @Test
> > -  public void testSourcePrimitiveDisplayData() {
>
> Additionally the @NeedsRunner idea is not going to work right now. Look
> at
> https://builds.apache.org/job/beam_PreCommit_MavenVerify/org.apache.beam$beam-runners-direct-java/2128/testReport/junit/org.apache.beam.sdk.transforms/CombineTest/testCombinePerKeyPrimitiveDisplayData/
> , where Scott tried this.
>
> The underlying assumption in the test is that the graph structure won't
> change, but it might. Until we have a better handle on the runner API,
> proper representation of composite transforms, and the requirements on
> runners to copy and move around display data when implementing overrides,
> I'm not sure we know how to write such tests.
>
> The failure linked above is checking for the exact classname used. But a
> runner implementing an override might substitute in a different class, for
> example. Seems like a fundamental issue.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/incubator-beam/pull/584/files/d9f5e3bdb641173fd776bde509d136f860f45616#r70090410>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AJnK7BNEH78JNxMChmd3WHoVj55uFd60ks5qTmmsgaJpZM4JDvGo>
> .
>